### ISSUE TITLE
Seeking Help for Understanding Code

### ISSUE THREAD

**Issue by SamuelGong** [HKUNLP/Dream#49]
I saw `logits = torch.cat([logits[:,:1], logits[:, :-1]], dim=1)`, and I did not know what it is for even if there is a comment as follows.

https://github.com/DreamLM/Dream/blob/3026ee0d9a11ee491bb34a1cda415fdde9094611/eval/eval.py#L353

If you really do not want the first logits, why copying it to make even one more?
